### PR TITLE
Save ID of selected medical issue

### PR DESCRIPTION
**Pull request by matrix-amiel**
Hi

I've found that the Id of medical issues is not saved in the lists table, only the title is saved.
This is problematic because the title can change and the diagnosis of patient can be lost.
I saved the ID in 'external_id' column to allow detection of specific medical issue for patient. 

Thanks
Amiel

### CODE REVIEW

**Comment by bradymiller** [comment]
hi @matrix-amiel ,
Will this step on any toes (ie. I am guessing the external_id was placed there for another reason/feature)?
-brady

**Comment by sjpadgett** [comment]
I'm pretty sure the external Id's used here and there are to relate items(procedure,allergies,issue's e.t.c) from imports(CCR,CCD) to their outside references. Note we have tables for this purpose such as external_encounters e.t.c

**Comment by matrix-amiel** [comment]
O.k. This was wrong idea to use external_id..
I fixed it and added new column instead.

File: sql/5_0_1-to-5_0_2_upgrade.sql [hunk 1]
```diff
@@ -94,3 +94,7 @@ INSERT INTO list_options ( list_id, option_id, title, seq, is_default ) VALUES (
 #IfNotRow2Dx2 list_options list_id proc_type option_id for title Custom Favorite Item
 INSERT INTO list_options ( list_id, option_id, title, seq, is_default ) VALUES ('proc_type','for','Custom Favorite Item' ,60,0);
 #EndIf
+
+#IfMissingColumn lists list_option_id
+ALTER TABLE `lists` ADD `list_option_id` VARCHAR (100) DEFAULT NULL COMMENT 'Reference to list_options table' AFTER `external_id`;
```
[the remark this answers is not in the document]
Comment (matrix-amiel) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: done

**Comment by bradymiller** [comment]
hi @matrix-amiel ,
Code looks good and will bring this into codebase. Thanks for the improvement! -brady

**Pull request merged**